src/lang/queryAst.ts

```diff
@@ -792,3 +795,43 @@ export function getObjExprProperty(
   if (index === -1) return null
   return { expr: node.properties[index].value, index }
 }
+
+/**
+ * Given an AST, returns the settings annotation object if it exists.
+ */
+export function getSettingsAnnotation(
```

Comment:
Instead of this TS function, we couldn't use `get_meta_settings()` from https://github.com/KittyCAD/modeling-app/pull/5200?

Reply:
Oh yes of course, sorry I wrote this TS function first and didn't revisit it, will do thank you. I think I just need to expose it

Reply:
Yeah the function exposed to TS is not `change_meta_settings`, it's `change_kcl_settings`, so I'll have to make a little wasm_bindgen version called `get_kcl_settings` with a similar setup. Shouldn't be too hard.

Reply:
Hm, `get_meta_settings` is not exposed to TS from what I can see. I think I could make one that takes a KCL string like `change_kcl_settings` does but that would require re-parsing, and we have already parsed the program when I go to use `getSettingsAnnotation`. I feel like I'm missing something

Reply:
I can help you with it on Monday. It could accept the Program similar to `executor()`. In the future, maybe we can add a parse cache similar to the way we have an execution cache.

Reply:
Pushed.